
+++ //depot/CommEngine/patches/WLAN/PATCH_502L07_P1/data-model/unfwlcfg/value-mapper-tr98.xml
--- //depot/CommEngine/patches/WLAN/PATCH_502L07_P1/data-model/unfwlcfg/value-mapper-tr98.xml

@@ -16,6 +16,7 @@
 <mapper name="WLBAND." shortObjectName="WLBAND" requirements="P"  />
   <value param="2.4GHz" nvram=2 enum="BAND_B" />
   <value param="5GHz"   nvram=1 enum="BAND_A" />
+  <value param="6GHz"   nvram=4 enum="BAND_6G" />
 
 
 <mapper name="WLREGMODE." shortObjectName="WLREGMODE" requirements="P" supportLevel="Present"  />

+++ //depot/CommEngine/patches/WLAN/PATCH_502L07_P1/userspace/private/libs/wlmdm/src/chanspec.c
--- //depot/CommEngine/patches/WLAN/PATCH_502L07_P1/userspace/private/libs/wlmdm/src/chanspec.c

@@ -645,7 +645,7 @@
         bw = chanspec_to_bw_str(chspec);
 
         /* ctl sideband string instead of BW for 40MHz */
-        if (chspec->band_width == WL_CHANSPEC_BW_40)
+        if (chspec->band_width == WL_CHANSPEC_BW_40 && chspec->band != WL_CHANSPEC_BAND_6G)
         {
             sb = chspec_sb_upper(chspec) ? "u" : "l";
             snprintf(buf, size, "%s%d%s", band, ctl_chan, sb);

+++ //depot/CommEngine/patches/WLAN/PATCH_502L07_P1/userspace/private/libs/wlmdm/src/chanspec.h
--- //depot/CommEngine/patches/WLAN/PATCH_502L07_P1/userspace/private/libs/wlmdm/src/chanspec.h

@@ -146,9 +146,9 @@
 enum WL_CHANSPEC_BAND
 {
     WL_CHANSPEC_BAND_2G = 0,
-    WL_CHANSPEC_BAND_6G = 1,
-    WL_CHANSPEC_BAND_4G = 2,
-    WL_CHANSPEC_BAND_5G = 3
+    WL_CHANSPEC_BAND_4G = 1,
+    WL_CHANSPEC_BAND_5G = 2,
+    WL_CHANSPEC_BAND_6G = 3
 };
 
 typedef enum WL_CHANSPEC_BW {

+++ //depot/CommEngine/patches/WLAN/PATCH_502L07_P1/userspace/private/libs/wlmdm/src/tr98_special.c
--- //depot/CommEngine/patches/WLAN/PATCH_502L07_P1/userspace/private/libs/wlmdm/src/tr98_special.c

@@ -388,9 +388,13 @@
             {
                 chanspec->band = WL_CHANSPEC_BAND_5G;
             }
+            else if (0 == strcmp((char *)buf_1, "2"))
+            {
+                chanspec->band = WL_CHANSPEC_BAND_2G;
+            }
             else
             {
-                chanspec->band = WL_CHANSPEC_BAND_2G;
+                chanspec->band = WL_CHANSPEC_BAND_6G;
             }
         }
     }

+++ //depot/CommEngine/patches/WLAN/PATCH_502L07_P1/userspace/private/libs/wlmdm/src/wifi_constants.h
--- //depot/CommEngine/patches/WLAN/PATCH_502L07_P1/userspace/private/libs/wlmdm/src/wifi_constants.h

@@ -65,7 +65,8 @@
 typedef enum wifi_bands
 {
     BAND_A = 1,
-    BAND_B = 2
+    BAND_B = 2,
+    BAND_6G = 4
 } WIFI_BANDS;
 
 typedef enum wifi_bandwidths
